### ISSUE TITLE
Connection/authentication problems with qBittorrent

### ISSUE THREAD

**Issue by coaperator** [tympanix/Electorrent#201]
**Client:**
- [ ] µTorrent
- [x] qBittorrent
- [ ] Transmission
- [ ] rTorrent
- [ ] Deluge

**Client Version:**
3.00 latest

**Operating System:**
win x64

**Application Version:**
2.7.1 latest

**Description:**
the same story as here https://github.com/tympanix/Electorrent/issues/196
![image](https://user-images.githubusercontent.com/6645911/102147617-17237a80-3e7c-11eb-82d6-923067fef37a.png)

**Comment by DynaZor** [comment]
Just jumping in to try to help.
Two issues could be the cause of the issue, according to the screenshot:

- The IP Address of the server seems like the Router's IP, make sure this is the right IP Address of the client's machine.
- No Password - Maybe try setting a password for QBittorrent's web interface other than null (empty) and match it in Electorrent's password field.

Also, check the port is the right port of the client's Web Interface port.

I hope this is just a mistake rather than a bug,
please update if anything changes and what you did to do so.

<3 DynaZor

**Comment by tympanix** [comment]
Empty passwords is a known bug. There's a fix on the way for this. Can you confirm that it works with a password. Also double check your connections options

**Comment by coaperator** [comment]
> Empty passwords is a known bug. There's a fix on the way for this. Can you confirm that it works with a password. Also double check your connections options

tried with a password, just a screenshot came out
there is no more possibility to test, I have already removed qBittorrent from the router
if there is time on the weekend, I'll try

**Comment by tympanix** [comment]
You mention you install qBittorrent on your router. This might be the difference which cause the problem. Try installing qBittorrent on your PC. Else you will have to troubleshoot your BitTorrent installation on your router

**Comment by leni8ec** [comment]
Same here - doesn't connect to the qbittorrent client on the router...

Electorrent: v2.8.0
qbittorrent: v4.4.3.1

![image](https://user-images.githubusercontent.com/2379473/193672059-cde2f7a9-529f-45d9-a13b-d85226a5a297.png)

**Comment by tympanix** [comment]
Hi @leni8ec

This is a qBittorrent installation on a router, yes? Can you explain the setup and which router that is? Did this work before 2.8.0?

What is the URL you use to connect to qBittorrent in the browser (Chrome, Firefox etc.)? Are the settings in Electorrent the same as this URL?

**Comment by leni8ec** [comment]
Hi @tympanix, yes - qBittorrent installed on router.
Tested Electorrent versions 2.7.3 and 2.8.0 - same result.

Router: [Keenetic Ultra KN-1810](https://keenetic.com/global/keenetic-ultra). 
There is support for [opkg](https://help.keenetic.com/hc/ru/articles/360000948719-OPKG), through which qbitorrent was installed.

Connect Electorrent to Transmission on this router - is OK , but not for qBittorrent.

Web-UI Url: `http://192.168.1.1:9080`

![image](https://user-images.githubusercontent.com/2379473/193901131-f685af2d-de16-4fd0-b005-2d600bcfab50.png)

**Comment by tympanix** [comment]
@leni8ec thank you for the detailed information. Your information looks correct.

I'm not sure how we're going to approach this problem. I will have a hard time replicating your environment to debug this particular error. This leaves me pretty much at guessing at the root cause, which will not be a viable approach. Do you have any suggestions on how to proceed with this?

**Comment by leni8ec** [comment]
@tympanix Yes, I understand that it is difficult to reproduce the problem in this case.

I don't know much about networking protocols, so I can't tell you more about it, unfortunately.

**Comment by tympanix** [comment]
If you get the chance (and would like to) to sniff the traffic with a tool like Wireshark from both the qBittorrent webui and Electorrent to compare, I would happily look through it. However, make sure the dump does not contains sensitive information and is shared securely (not publicly)

**Comment by dog1986** [comment]
Hi😊 I have a same problem😊 qbittorent 4.4.5 on my router [https://keenetic.com/en/keenetic-hero](url) installed opkg Same Web-UI Url: http://192.168.1.1:9080
Do you have solution?

**Comment by tympanix** [comment]
@dog1986 we need someone with a Keenetic router to produce a working example of the problem or send some network diagnostics (i.e. using Wireshark)

**Comment by tympanix** [comment]
Same as #369.

Fixed in #614
